pkg/sensors/tracing/options.go

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Tetragon
+
+package tracing
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/cilium/tetragon/pkg/k8s/apis/cilium.io/v1alpha1"
+	"github.com/cilium/tetragon/pkg/logger"
+	"github.com/cilium/tetragon/pkg/option"
+)
+
+type opt struct {
+	name string
+	set  func(val string) error
+}
+
+type kprobeOptions struct {
+	DisableKprobeMulti bool
+}
+
+func getKprobeOptions(specs []v1alpha1.OptionSpec) (*kprobeOptions, error) {
+	options := &kprobeOptions{}
+
+	// Allowed options
+	var opts = []opt{
+		opt{
+			// local --disable-kprobe-multi
+			name: option.KeyDisableKprobeMulti,
+			set: func(str string) (err error) {
+				options.DisableKprobeMulti, err = strconv.ParseBool(str)
+				return err
+			},
+		},
+	}
+
+	// cross reference allowed with passed options
+	for i := range specs {
```

Comment:
nit: You can use something like:
```
	for _, spec := range specs {
		for _, opt := range opts {
```
to remove the need to declare new variables.

Reply:
You can also have a global variable `opts` instead of having to initialize that every time we call that function. Additionally, you can use a map (key: name, value: set) and remove the need for the nested loop. 

(feel free to ignore these comments as this is not a performance-critical code, but I believe that these changes will make the code a bit easier to read)

Reply:
yes, it's better with the map, will change that, thanks